phoenix-core/src/main/java/org/apache/phoenix/parse/PhoenixRowTimestampParseNode.java

```diff
@@ -0,0 +1,89 @@
+import org.apache.phoenix.expression.Expression;
+import org.apache.phoenix.expression.KeyValueColumnExpression;
+import org.apache.phoenix.expression.function.FunctionExpression;
+import org.apache.phoenix.expression.function.PhoenixRowTimestampFunction;
+import org.apache.phoenix.query.QueryConstants;
+import org.apache.phoenix.schema.PDatum;
+import org.apache.phoenix.schema.PTable;
+import org.apache.phoenix.schema.SortOrder;
+import org.apache.phoenix.schema.types.PDataType;
+import org.apache.phoenix.schema.types.PDate;
+import org.apache.phoenix.util.IndexUtil;
+import org.apache.phoenix.util.SchemaUtil;
+
+import java.sql.SQLException;
+import java.util.Arrays;
+import java.util.List;
+
+public class PhoenixRowTimestampParseNode extends FunctionParseNode {
+
+    PhoenixRowTimestampParseNode(String name, List<ParseNode> children,
+                                 BuiltInFunctionInfo info) {
+        super(name, children, info);
+    }
+
+    @Override
+    public FunctionExpression create(List<Expression> children, StatementContext context) throws SQLException {
+
+        // PhoenixRowTimestampFunction does not take any parameters.
+        assert children.size() == 0;
+
+        // Get the empty column family and qualifier for the context.
+        PTable table = context.getCurrentTable().getTable();
+        byte[] emptyColumnFamilyName = SchemaUtil.getEmptyColumnFamily(table);
+        byte[] emptyColumnName = table.getEncodingScheme() == PTable.QualifierEncodingScheme.NON_ENCODED_QUALIFIERS ?
+                QueryConstants.EMPTY_COLUMN_BYTES :
+                table.getEncodingScheme().encode(QueryConstants.ENCODED_EMPTY_COLUMN_NAME);
+
+        // Create an empty column key value expression.
+        // This will cause the empty column key value to evaluated during scan filter processing.
+        List<Expression> emptyColumnExpression = Arrays.asList(new Expression[] {new KeyValueColumnExpression(new PDatum() {
+            @Override
+            public boolean isNullable() {
+                return true;
+            }
+            @Override
+            public PDataType getDataType() {
+                return PDate.INSTANCE;
+            }
+            @Override
+            public Integer getMaxLength() {
+                return null;
+            }
+            @Override
+            public Integer getScale() {
+                return null;
+            }
+            @Override
+            public SortOrder getSortOrder() {
+                return SortOrder.getDefault();
+            }
+        }, emptyColumnFamilyName, emptyColumnName)});
+
+        // Add the empty column to the projection list.
+        // According to PHOENIX-4179 this will then return the timestamp of the empty column cell.
+        IndexUtil.addEmptyColumnToScan(context.getScan(), emptyColumnFamilyName, emptyColumnName);
+        return new PhoenixRowTimestampFunction(emptyColumnExpression, emptyColumnFamilyName, emptyColumnName);
+    }
+}
```

Comment:
@jpisaac let's add unit tests for the newly added classes, however small the test classes may be. Good for code coverage.

Reply:
Added it the unit tests